workers/unity/Packages/com.improbable.gdk.core/Worker/DefaultWorkerConnector.cs

```diff
@@ -10,62 +12,151 @@ public abstract class DefaultWorkerConnector : WorkerConnector
         /// </summary>
         public bool UseExternalIp;
 
-        /// <summary>
-        ///     Determines whether to connect via the locator.
-        /// </summary>
-        /// <returns>True, if should connect via the Locator, false otherwise.</returns>
-        protected override bool ShouldUseLocator()
+        protected override ConnectionService GetConnectionService()
         {
+            var commandLineArguments = Environment.GetCommandLineArgs();
+            var commandLineArgs = CommandLineUtility.ParseCommandLineArgs(commandLineArguments);
+
+            if (commandLineArgs.ContainsKey(RuntimeConfigNames.SteamDeploymentTag) ||
+                commandLineArgs.ContainsKey(RuntimeConfigNames.SteamTicket))
+            {
+                return ConnectionService.Locator;
+            }
+
+            if (commandLineArgs.ContainsKey(RuntimeConfigNames.LoginToken))
+            {
+                return commandLineArgs.ContainsKey(RuntimeConfigNames.PlayerIdentityToken)
+                    ? ConnectionService.AlphaLocator
+                    : ConnectionService.Locator;
+            }
+
+            return ConnectionService.Receptionist;
+        }
+
+        protected override ConnectionParameters GetConnectionParameters(string workerType, ConnectionService service)
+        {
+            // UseExternalIp needs to be true when using the locator
+            var useExternalIp = service != ConnectionService.Receptionist || UseExternalIp;
```

Comment:
This should be `service == this || service == that` instead of `service != other /* this imples it's this or that */` (if we add a new method it'll be a bug)

Reply:
good point!